File: 1684_consistent_strings.cpp
#include <iostream>

using namespace std;

int get_repr(string word) {
    int repr = 0;
    for(auto alpha:word) {
        repr |= 1 << (alpha - 'a');
    }
    return repr;
}

int countConsistentStrings(string allowed, vector<string>& words) {
    int alpha = ~get_repr(allowed);
    int count = 0;

    for (auto word: words) {
        int repr = get_repr(word);
        if (alpha & repr) count++;
    }
    return words.size() - count;
    
}

int main() {
    vector<string> words = {"ad","bd","aaab","baa","badab"};
    string allowed = "ab";
    
    cout << countConsistentStrings(allowed, words) << endl;


}

File: 1684_consistent_strings.py
def countConsistentStrings(allowed, words):
    def get_repr(s):
        repr = 0
        for alpha in s:
            repr[ord(alpha) - ord('a')] = 1
        return tuple(repr)

    count = 0
    allowed = get_repr(allowed)
    for word in words:
        count += allowed == get_repr(word)
    return count


print(countConsistentStrings("ab", ["ad", "bd", "aaab", "baa", "badab"]))
print(countConsistentStrings("abc", ["a", "b", "c", "ab", "ac", "bc", "abc"]))
